modin/data_management/functions/groupby_function.py

```diff
@@ -148,8 +148,9 @@ def caller(
         numeric_only=True,
         **kwargs,
     ):
-        if not (isinstance(by, (type(query_compiler)) or hashable(by))) or isinstance(
-            by, pandas.Grouper
+        if groupby_args.get("level", None) is None and (
```

Comment:
probably a typo here:
```suggestion
        if groupby_args.get("level", None) is not None and (
```
otherwise, we're defaulting to pandas at any time when `by` is specified

Reply:
I don't think that is right. Then we would always default to pandas if `level` is specified. It does not default to pandas when `by` is specified in this logic, we add a clause here that says that if `level` is None and all other conditions are met we default.

Reply:
oh, got it

Reply:
@devin-petersohn well, actually I don't got it)
what does `if level is None` check mean in this context? This if-statement routes us either to default to pandas (if arguments are too complex and we don't support them now) or to Modin implementation. It seems that this added line is redundant:
- if the passed type of `by` is not supported then we definitely should go to "default_to_pandas" if-branch, but we're doing this extra check about `level` parameter.
- if the passed type of `by` is supported we will go to Modin groupby implementation anyway, so the `level` check is also unnecessary. 

Could you please explain why this check is needed? I've removed this check locally and all of the tests passed (dataframe + series + groupby tests).

Reply:
`level` will default to pandas if this line is not included. It would be expected to pass.

if `level is not None`, then `by is None`, and the code will default to pandas. This check will bypass the checks on `by` if `level is not None`.

Reply:
this code will not default to pandas in case of `by=None` since `None` is hashable

Reply:
probably you thought that it is, because previously there was a typo which was mentioned [here](https://github.com/modin-project/modin/pull/2758#discussion_r585156479), that causes us to default to pandas is `by` is hashable